cloud/cloud_connection_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCloudConnection(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testCheckcloudConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceDataSourcecloudConnection(
+					"sndev",
+					"terraform-test-cloud-connection-b",
+					"us-west1",
+					"sn-cloud-connection"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckcloudConnectionExists("streamnative_cloud_connection.test-cloud-connection"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckcloudConnectionDestroy(s *terraform.State) error {
+	// Add a sleep for wait the service account to be deleted
+	// It seems that azure connection to gcp is slow, so add a delay to wait
+	// for the resource to be cleaned up and check it again
```

Comment:
Why need wait for service account to be deleted?

Reply:
Fixed, was a copy from the service account test I missed removing